buildpipeline/Core-Setup-Linux-BT.json

```diff
@@ -526,6 +526,96 @@
         "failOnStandardError": "false"
       }
     },
+	{
+      "enabled": true,
+      "continueOnError": false,
+      "alwaysRun": false,
+      "displayName": "Initialize docker - Debian 9",
```

Comment:
```
"arguments": "run --rm $(DockerCommonRunArgs_Debian9) /bin/bash -c \"HOME=$(PB_GitDirectory); git clean -X -d -f; $(PB_GitDirectory)/init-tools.sh\"",
```

The first step you run is `git clean` which to me appears to be cleaning out the entire repo and then rebuilding. Perhaps I've missed where you are creating another clone but it certainly looks like you are using the same clone.